1-Draft/RFCXXXX-RFC-Invoke-DscResource.md

```diff
@@ -0,0 +1,126 @@
+---
+RFC: RFCXXXX
+Author: Gael Colas
+Status: Draft
+SupersededBy: N/A
+Version: 0.6
+Area: Microsoft.PowerShell.DesiredStateConfiguration
+---
+
+# Invoke-DscResource
+
+Add support for `Invoke-DscResource` in PowerShell 7+.
+
+This RFC addresses the need to leverage the DSC ecosystem of resources from newer versions of PowerShell, the way to decouple the execution of resources from the LCM and CIM/WMI, and the user experience from a consumer and solution vendor/integrator point of view.
+
+## Motivation
+
+1. DSC has been cited by users and solution partners as a blocker for moving from Windows PowerShell to PowerShell 7+.
+
+2. As part of the latest survey on DSC usage started in June 2019, the top requested feature was support for Invoke-DscResource in PSCore without WMI dependency.
+
+3. The DSC Ecosystem and Community would greatly benefit from enabling DSC resources to be used in imperative scripts, in current user context.
+
+## Specification
+
+### Invoke-DscResource in a module, decoupled from PowerShell
+
+`Invoke-DscResource` is directly related to `Get-DscResource` in terms of user experience: Get provides the discoverability before the invocation.
+
+`Invoke-DscResource` does not need to be part of the PowerShell engine (same as `Get-DscResource`), and we should aim to decouple PowerShell's engine from the DSC Ecosystem. While it's convenient for the `Configuration` keyword to have some of its implementation done in PowerShell's engine for parsing, `Invoke-DscResource` has no such requirements and should live in an independent module evolving in its own timeline, that will be bundled and shipped with PowerShell releases.
+
+Open-sourcing this module should be a priority, but is out of scope for this RFC.
```

Comment:
I would think this work is done in PSDesiredStateConfiguration and be OSS.  Doesn't seem like this line is necessary.

Reply:
PSDesiredStateConfiguration isn’t currently OSS

Reply:
It's not OSS currently OSS and probably can't be until we put that in its own pipeline. That's something I want to discuss with Nitin an Amit.

Reply:
Yeah, forgot about that.  This line makes sense then.

Reply:
@stevel-msft this is something I didn’t think of in my estimate.

Reply:
We'll leave it to the DSC team to do the work to make it OSS.

Reply:
There are two parts of this.  The engineering system for the repo, and actually Open Sourcing it.  Without at least a basic engineering system, we should not open source this or add a feature like this to the current module.  We need to resolve this.

Reply:
This is something we will plan in the future.

Reply:
Nitin has agreed to committing to making this OSS in the next milestone after the current one.  So consider this closed.

Reply:
Great!